bin/annotate-genome.py

```diff
@@ -0,0 +1,195 @@
+            aend = alignmentEnd(
+                alignment.referenceAligned.sequence,
+                gappedOffset,
+                feature["stop"] - feature["start"],
+            )
+            resultFeatures[name]["debug"] = {
+                "genome orf": orf,
+                "feature alignment": {
+                    "reference": alignment.referenceAligned.sequence[gappedOffset:aend],
+                    "genome": alignment.genomeAligned.sequence[gappedOffset:aend],
+                },
+                "reference feature": feature.copy(),
+            }
+
+        if args.reportDifferences:
+            if not orf["foundStartCodon"]:
+                print(
+                    f"No START codon found for genome feature " f"{feature['name']!r}.",
+                    file=sys.stderr,
+                )
+
+            if not orf["foundStopCodon"]:
+                print(
+                    f"No STOP  codon found for genome feature " f"{feature['name']!r}.",
+                    file=sys.stderr,
+                )
+
+    # Sanity check that the Features class can load what we just produced.
+    Features(result, sars2=args.sars2)
+    print(dumps(result, sort_keys=True, indent=4))
+
+    if args.summarize:
+        print(f'Summary of changes for {result["id"]!r}', file=sys.stderr)
+        count = 0
+        for name, info in result["features"].items():
+            try:
+                aaDiffs = info["aa differences"]["substitutions"]
+            except KeyError:
+                aaDiff = "-"
+            else:
+                aaDiff = ", ".join(aaDiffs)
+
+            try:
+                n = info["aa differences"]["lengths"]["difference"]
+            except KeyError:
+                lenDiff = "-"
+            else:
+                lenDiff = f"{abs(n)} aa " + ("shorter" if n < 0 else "longer")
+
+            if aaDiff or lenDiff:
+                count += 1
+                print(
+                    "\t".join(
+                        (str(count), name, aaDiff, lenDiff, info.get("note", ""))
+                    ),
+                    file=sys.stderr,
+                )
+
+    return 0
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Print JSON annotations for a genome (or genomes).",
+    )
+
+    parser.add_argument(
+        "--genome",
+        metavar="file.fasta",
+        type=argparse.FileType("r"),
+        default=sys.stdin,
+        help="The FASTA file containing the SARS-CoV-2 genome(s) to examine.",
+    )
+
+    parser.add_argument(
+        "--summarize",
+        action="store_true",
+        help="Write a summary of differences to standard error.",
+    )
+
+    parser.add_argument(
+        "--reportDifferences",
+        action="store_true",
+        help=(
+            "If the genome ORF differs from the reference due to no start "
+            "or stop codon, print a message.",
```

Comment:
Do you really need this trailing comma?

Reply:
Huh - thanks! I guess that help string was shorter once and had a comma after it. then I made it longer and put the parens in wrongly and accidentally created a `tuple`.  I've started trying to use typing hints and checking the code with `mypy` but there are still too many warnings to deal with so I'm going slowly and not running `mypy` all the time. Had I done so, I would perhaps have seen that it says:

```
bin/annotate-genome.py:184: error: Argument "help" to "add_argument" of "_ActionsContainer" has incompatible type "Tuple[str]"; expected "Optional[str]"
```

Anyway, thanks :-)